ToolBox/IEX/IEXDataQueueHandler.cs

```diff
@@ -418,95 +384,124 @@ private IEnumerable<Slice> ProcessHistoryRequests(Data.HistoryRequest request)
-                    begin = begin.AddDays(1);
+                    var begin = start;
+                    while (begin < end)
+                    {
+                        var url =
+                            $"{baseUrl}/{ticker}/chart/date/{begin.ToStringInvariant("yyyyMMdd")}?token={_apiKey}";
+                        urls.Add(url);
+                        begin = begin.AddDays(1);
+                    }
+
+                    break;
+                }
+                case Resolution.Daily:
+                {
+                    string suffix;
+                    if (span.Days < 30)
+                    {
+                        suffix = "1m";
+                    }
+                    else if (span.Days < 3 * 30)
+                    {
+                        suffix = "3m";
+                    }
+                    else if (span.Days < 6 * 30)
+                    {
+                        suffix = "6m";
+                    }
+                    else if (span.Days < 12 * 30)
+                    {
+                        suffix = "1y";
+                    }
+                    else if (span.Days < 24 * 30)
+                    {
+                        suffix = "2y";
+                    }
+                    else if (span.Days < 60 * 30)
+                    {
+                        suffix = "5y";
+                    }
+                    else
+                    {
+                        suffix = "max";   // max is 15 years
+                    }
+
+                    var url =
+                        $"{baseUrl}/{ticker}/chart/{suffix}?token={_apiKey}";
+                    urls.Add(url);
+
+                    break;
                 }
-            }
-            else if (span.Days < 30)
-            {
-                suffixes.Add("1m");
-            }
-            else if (span.Days < 3 * 30)
-            {
-                suffixes.Add("3m");
-            }
-            else if (span.Days < 6 * 30)
-            {
-                suffixes.Add("6m");
-            }
-            else if (span.Days < 12 * 30)
-            {
-                suffixes.Add("1y");
-            }
-            else if (span.Days < 24 * 30)
-            {
-                suffixes.Add("2y");
-            }
-            else
-            {
-                suffixes.Add("5y");
             }
 
             // Download and parse data
-            using (var client = new System.Net.WebClient())
+            var requests = new List<Task<string>>();
+
+            urls.DoForEach(url =>
             {
-                foreach (var suffix in suffixes)
+                using (var client = new WebClient())
                 {
-                    var response = client.DownloadString("https://cloud.iexapis.com/v1/stock/" + ticker + "/chart/" + suffix + "?token=" + _apiKey);
-                    var parsedResponse = JArray.Parse(response);
+                    requests.Add(client.DownloadStringTaskAsync(new Uri(url)));
```

Comment:
It seems there is no harm in current implementation : https://stackoverflow.com/questions/39609125/object-disposal-and-garbage-collection-prior-to-event-triggering. Since web clients dispose() doesn't not dispose any resources that would be used by DownloadStringTaskAsync(). And clients itself is not distroyed by GC: https://stackoverflow.com/questions/421547/does-the-garbage-collector-destroy-temporarily-unreferenced-objects-during-async but to be on a safe side I am chaning impl to call dispose after async call is finished and result is obtained..